ibis/common/temporal.py

```diff
@@ -114,6 +119,72 @@ class IntervalUnit(TemporalUnit):
     NANOSECOND = "ns"
 
 
+def normalize_timedelta(
+    value: datetime.timedelta | numbers.Real, unit: IntervalUnit
+) -> datetime.timedelta:
+    """Normalize a timedelta value to the given unit.
+
+    Parameters
+    ----------
+    value
+        The value to normalize, either a timedelta or a number.
+    unit
+        The unit to normalize to.
+
+    Returns
+    -------
+    The normalized timedelta value.
+
+    Examples
+    --------
+    >>> from datetime import timedelta
+    >>> normalize_timedelta(1, IntervalUnit.SECOND)
+    1
+    >>> normalize_timedelta(1, IntervalUnit.DAY)
+    1
+    >>> normalize_timedelta(timedelta(days=1), IntervalUnit.SECOND)
+    86400
+    >>> normalize_timedelta(timedelta(seconds=3), IntervalUnit.MILLISECOND)
+    3000
+    >>> normalize_timedelta(timedelta(seconds=3), IntervalUnit.MICROSECOND)
+    3000000
+    """
+    if isinstance(value, datetime.timedelta):
+        # datetime.timedelta only stores days, seconds, and microseconds internally
+        total_seconds = value.total_seconds()
+        if unit == IntervalUnit.NANOSECOND:
+            value = total_seconds * 1e9
+        elif unit == IntervalUnit.MICROSECOND:
+            value = total_seconds * 1e6
+        elif unit == IntervalUnit.MILLISECOND:
+            value = total_seconds * 1e3
+        elif unit == IntervalUnit.SECOND:
+            value = total_seconds
+        elif unit == IntervalUnit.MINUTE:
+            value = total_seconds / 60
+        elif unit == IntervalUnit.HOUR:
+            value = total_seconds / 3600
+        elif unit == IntervalUnit.DAY:
+            value = total_seconds / 86400
```

Comment:
86400 is always the length of a day. What about deltas between timezone changes?